src/actions/airtable/airtable.ts

```diff
@@ -55,32 +55,32 @@ export class AirtableAction extends Hub.OAuthAction {
       let accessToken
       if (request.params.state_json) {
         const stateJson = JSON.parse(request.params.state_json)
-        const refreshResponse = await this.refreshTokens(stateJson.tokens.refresh_token)
-        accessToken = (refreshResponse as any).data.access_token
-        // Every single access_token invalidates previous refresh_token. Need to
-        // update state on EVERY request
+        accessToken = stateJson.tokens.access_token
         state.data = JSON.stringify({
           tokens: {
-            refresh_token: (refreshResponse as any).data.refresh_token,
+            refresh_token: stateJson.tokens.refresh_token,
             access_token: accessToken,
           },
         })
       }
-      const airtableClient = await this.airtableClientFromRequest(accessToken)
-      const base = airtableClient.base(request.formParams.base)
-      const table = base(request.formParams.table)
 
-      await Promise.all(records.map(async (record: any) => {
-        return new Promise<void>((resolve, reject) => {
-          table.create(record, (err: { message: string } | null, rec: any) => {
-            if (err) {
-              reject(err)
-            } else {
-              resolve(rec)
-            }
+      try {
+        await this.executeAirtable(request, records, accessToken)
+      } catch {
```

Comment:
Essentially using an exception as an if-else feels wrong in this context. Can we try to make this a flow that does not exceptions instead?

Reply:
It's effectively more like a retry - I could rewrite it by making the logic fit inside a sub method with a retry count but since I wanted to get something done super quick I just did it like this.

The condition to trigger this is that the external call fails due to oauth failure, so then we trigger the refresh workflow.